gradle-plugin/kubernetes/src/main/java/org/eclipse/jkube/gradle/plugin/task/KubernetesHelmPushTask.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (c) 2019 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at:
+ *
+ *     https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.jkube.gradle.plugin.task;
+
+import org.eclipse.jkube.gradle.plugin.KubernetesExtension;
+import org.eclipse.jkube.kit.resource.helm.HelmConfig;
+
+import javax.inject.Inject;
+
+import static org.eclipse.jkube.kit.resource.helm.HelmServiceUtil.initHelmConfig;
+import static org.eclipse.jkube.kit.resource.helm.HelmServiceUtil.initHelmPushConfig;
+
+public class KubernetesHelmPushTask extends AbstractJKubeTask {
+  @Inject
+  public KubernetesHelmPushTask(Class<? extends KubernetesExtension> extensionClass) {
+    super(extensionClass);
+    setDescription("Upload a helm chart to specified helm repository.");
+  }
+
+  @Override
+  public void run() {
+    if (kubernetesExtension.getSkipOrDefault()) {
+      return;
+    }
+    try {
+      HelmConfig helm = initHelmConfig(kubernetesExtension.getDefaultHelmType(), kubernetesExtension.javaProject,
+        kubernetesExtension.getKubernetesManifestOrDefault(), kubernetesExtension.getKubernetesTemplateOrDefault(),
+        kubernetesExtension.helm).build();
```

Comment:
Is this because here the task doesn't extend HelmPushTask (as opposed to HelmPushMojo that does extend HelmMojo)?

Reply:
yes

Reply:
OK, we need to refactor those helpers.